### PR TITLE
[DOCS] How to introspect and partition an SQL database

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-200/GREAT-222/GREAT-288
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+<Prerequisites>
+
+- [Configured and loaded a Data Context](../../tutorials/getting_started/initialize_a_data_context.md)
+- [Configured a Datasource and Data Connector](../../reference/datasources.md)
+  
+</Prerequisites>
+
+This guide will use the following `Datasource` configuration as an example:
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_introspect_and_partition_your_data/sql_database/yaml_example.py#L9-L66
+```
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md]: This is amazing. Thank you so much for the in-line comments ❤️  they help a TON

File: docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+Finally, confirm the expected number of batches was retrieved and the reduced size of a batch (due to sampling):
+
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_introspect_and_partition_your_data/sql_database/yaml_example.py#L170-L176
+```
+
+## Additional Notes
+
+Available `Splitting` methods and their configuration parameters:
+
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md]: ❤️

File: docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+`Splitting` is a `Data Connector` capability that provides the means of breaking the batch data up based on the values
+of certain dimensions of the data of interest.  Employing `Splitting` as part of introspection can help focus on smaller
+batches, centered around a common aspect of the data.
+
+To configure `Splitting`, specify a dimension (i.e., `column_name` or `column_names`), the method of splitting, and
+parameters to be used by the specified splitting method.
+
+For the present example, we can split according to the "pickup_datetime" column parsed to the date level precision and
+thus obtain batches containing the taxi rides data for each day found in the available data:
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md]: Wanting to check formatting. Should "pickup_datetime" be `pickup_datetime`?

**Review by Shinnnyshinshin — APPROVED** [review]
Love this. Minor non-blocking comment regarding formatting. LGTM

File: docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md (hunk 1 above)
Comment (alexsherstinsky) [reply docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md]: 🙇

File: docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md (hunk 3 above)
Comment (alexsherstinsky) [reply docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md]: Fixed!  Nice catch @Shinnnyshinshin !  Thanks!

**Review by anthonyburdi — APPROVED** [review]
LGTM!

File: docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+Finally, confirm the expected number of batches was retrieved and the reduced size of a batch (due to sampling):
+
+```python file=../../../tests/integration/docusaurus/connecting_to_your_data/how_to_introspect_and_partition_your_data/sql_database/yaml_example.py#L170-L176
+```
+
+## Additional Notes
+
+Available `Splitting` methods and their configuration parameters:
+
+    +-----------------------------------+---------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	+
```
Comment (anthonyburdi) [inline docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md]: @alexsherstinsky I'm not sure why this isn't getting rendered into docusaurus format (I see it as a code block in the PR rendered output) but you might want to double check the syntax: https://www.markdownguide.org/extended-syntax/#tables - it also shows an example of the rendered table.

Comment (alexsherstinsky) [reply docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md]: @anthonyburdi Thank you for pointing this out.  I will be revisiting this document once the files one is merged (as the latter informs some of the document organization structure of this document as well).  Thanks!

File: docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+title: How to configure a DataConnector to introspect and partition tables in SQL
+---
+import Prerequisites from '../connecting_to_your_data/components/prerequisites.jsx'
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+This guide will help you introspect and partition tables in an SQL database using `SimpleSqlalchemyDatasource`, which
+operates as a proxy to `InferredAssetSqlDataConnector` and `ConfiguredAssetSqlDataConnector` (for the SQL use case,
+these types of the `Active Data Connector` are exercised indirectly, via `SimpleSqlalchemyDatasource`).  For background,
+please see the `Datasource` specific guides in the "Connecting to your data" section.
```
Comment (anthonyburdi) [inline docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md]: I think you can get rid of the parenthetical here, and leave the statement as-is.
```suggestion
This guide will help you introspect and partition tables in an SQL database using `SimpleSqlalchemyDatasource`, which
operates as a proxy to `InferredAssetSqlDataConnector` and `ConfiguredAssetSqlDataConnector`.  For background,
please see the `Datasource` specific guides in the "Connecting to your data" section.
```

File: docs/guides/connecting_to_your_data/advanced/how_to_configure_a_dataconnector_for_splitting_and_sampling_tables_in_sql.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
+---
+title: How to configure a DataConnector for splitting and sampling tables in SQL
+---
+import Prerequisites from '../../connecting_to_your_data/components/prerequisites.jsx'
+
+This guide will help you configure `Splitting` and `Sampling` for working with tables in an SQL database using
+`SimpleSqlalchemyDatasource`, which operates as a proxy to `InferredAssetSqlDataConnector` and
+`ConfiguredAssetSqlDataConnector` (for the SQL use case, these types of the `Active Data Connector` are exercised
+indirectly, via `SimpleSqlalchemyDatasource`).
```
Comment (anthonyburdi) [inline docs/guides/connecting_to_your_data/advanced/how_to_configure_a_dataconnector_for_splitting_and_sampling_tables_in_sql.md]: Same suggestion as the other doc (to remove the parenthetical).
```suggestion
This guide will help you configure `Splitting` and `Sampling` for working with tables in an SQL database using
`SimpleSqlalchemyDatasource`, which operates as a proxy to `InferredAssetSqlDataConnector` and
`ConfiguredAssetSqlDataConnector`.
```

File: docs/guides/connecting_to_your_data/advanced/how_to_configure_a_dataconnector_for_splitting_and_sampling_tables_in_sql.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+The `Splitting` and `Sampling` mechanisms provided by Great Expectations serve as additional tools for `Partitioning`
+your data at various levels of granularity:
+- `Splitting` provides the means of focusing the batch data on the values of certain dimensions of the data of interest.
+- `Sampling` provides a means for reducing the amount of data in the retrieved batch to facilitate data analysis.
+
+<Prerequisites>
+
+- [Configured and loaded a Data Context](../../../tutorials/getting_started/initialize_a_data_context.md)
+- [Configured a Datasource and Data Connector](../../../reference/datasources.md)
+- [How to configure a DataConnector to introspect and partition tables in SQL](../how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md)
```
Comment (anthonyburdi) [inline docs/guides/connecting_to_your_data/advanced/how_to_configure_a_dataconnector_for_splitting_and_sampling_tables_in_sql.md]:
```suggestion
- Reviewed [How to configure a DataConnector to introspect and partition tables in SQL](../how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md)
```

**Review by anthonyburdi — COMMENTED** [review]
Just a few small suggestions

File: docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md (hunk 5 above)
Comment (alexsherstinsky) [reply docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql.md]: Incorporated (in next commit).

File: docs/guides/connecting_to_your_data/advanced/how_to_configure_a_dataconnector_for_splitting_and_sampling_tables_in_sql.md (hunk 6 above)
Comment (alexsherstinsky) [reply docs/guides/connecting_to_your_data/advanced/how_to_configure_a_dataconnector_for_splitting_and_sampling_tables_in_sql.md]: Handled in upcoming commit.

File: docs/guides/connecting_to_your_data/advanced/how_to_configure_a_dataconnector_for_splitting_and_sampling_tables_in_sql.md (hunk 7 above)
Comment (alexsherstinsky) [reply docs/guides/connecting_to_your_data/advanced/how_to_configure_a_dataconnector_for_splitting_and_sampling_tables_in_sql.md]: Incorporated (in the next commit).

**Review by abegong — APPROVED** [review]
ToC changes look good. There's some other weirdness (see the video), but it's not my role to block the PR for that level of review.

https://www.loom.com/share/e188e43222b7421d83c454a417d4ff7a

**Review by Shinnnyshinshin — APPROVED** [review]
Looks wonderful :) Thank you @alexsherstinsky

**Comment by alexsherstinsky** [comment]
> Looks wonderful :) Thank you @alexsherstinsky

Thank you for re-review, @Shinnnyshinshin ! 🙇

**Review by talagluck — APPROVED** [review]
contributing.md LGTM!

**Review by austiezr — APPROVED** [review]

**Pull request merged**